docs/DeveloperGuide.md

```diff
@@ -1,33 +1,372 @@
+After the first launch, every time the user reopens the application, it will load the information to the Model from the local file. 
+Whenever the calendarList is updated, all information will be automatically saved to the same local file.
+
+### Model
+
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/Model_Class_Diagram.JPG" alt="" width="750"/>
+
+The figure given above shows the structure of the Model in this application. When executing commands, 
+the `CalendarItem` class updates the information or provides the information of different types of items if needed. 
+It is split into two subclasses: `Task` and `Event` class.
+The `CalendarList` class holds the array of `CalendarItem` and keeps track of the number of total items, total tasks, and total events.
+Given below is the simple overview of Task and Event classes.
+
+#### Task
+
+The `Task` class stores the information of all task items, including description and status, such as isDone and isImportant. 
+It has two subclasses: `Todo` and `Deadline`. `Deadline` items also have the date information and countdown for the deadline date, which are not included in `Todo` tasks. 
+
+#### Event
+
+The `Event` class stores the information of all `Event` items, including date, time, venue, status whether it is overdue, and any other information if added. 
+It has two subclasses: SchoolEvent and Activity. `Activity` items can have other details.
+All `SchoolEvent` items must have a module code.  
+Furthermore, the `SchoolEvent` class has four subclasses which are `Lecture`, `Tutorial`, `Lab`, and `Exam`. Among them only `Exam` items have a countdown for the exam date. 
+All `Event` items update the information or provide needed information about `Event` items when executing commands related to events or saving the information to `Storage`.
+
+Design consideration: `SchoolEvent` is modelled after NUS modules to cater to our intended users.
+
+## Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add a task item feature
+This feature is being facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:
+
+It first checks the type of the new task,
+then it analyses the attached information, if the format is desired, the program saves the event with the information in the task list.
+
+Note: A todo task has only task description while a deadline task has both the task description and a due date. 
+The due date must be in the pattern of ddMMyy as it is the pattern set by the `TimeParser` class.
+
+### Add an event item feature
+This feature is being facilitated by `AddCommand`.
+
+### Add a calendar item feature
+This feature is facilitated by `AddCommand`.
+The following sequence diagram shows how the `execute()` operation works:
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/add_command_SD.JPG" alt="" width="750"/><br/>
```

Comment:
The uml diagram here is slightly inconsistent with the ones below. Perhaps in a future version can startdardize the if using the alt box and object deletion parts